### PR TITLE
feat: add link component (resolves #12)

### LINKED ISSUE #12: Component: link
Implement links with all states.

## Additional context or notes

https://www.figma.com/file/LXnq7rdbsr2yyxOG2rSh5S/Inclusive-Design-Guide-2023?type=design&node-id=28-584&mode=design&t=WibM2Mq2lHqyiZx8-4

### PR DESCRIPTION
**Pull request by chosww**
See #12 for details.

### CODE REVIEW

File: src/_includes/components/link/link.config.js [hunk 1]
```diff
@@ -5,10 +5,24 @@ module.exports = {
...
         {
-            title: "External",
+            title: "White background",
             context: {
-                text: "An external link",
-                external: true
+                variantClass: "bg-white",
+                text: "Previous"
+            }
+        },
```
Comment (greatislander) [inline src/_includes/components/link/link.config.js:13]: The default is the white background, so you can remove this variant.

File: src/_includes/components/link/link.njk [hunk 2]
```diff
@@ -1 +1,8 @@
-<a href="https://guide.inclusivedesign.ca"{% if params.external %} rel="external"{% endif %}>{{ params.text }}</button>
+{% block head %}
+    <link rel="stylesheet" href="/static/css/link.css" />
+{% endblock %}
+
+
+<div {% if params.variantClass %} class="link-container {{ params.variantClass }}" {% endif %}>
+    <a href="#">{{ params.text }}</a>
+</div>
```
Comment (greatislander) [inline src/_includes/components/link/link.njk:8]:
```suggestion
{% if params.variantClass %}
<div  class="{{ params.variantClass }}" >
{% endif %}
    <a href="#">{{ params.text }}</a>
{%-if params.variantClass %}
</div>
{% endif %}
```

File: src/static/css/link.css [hunk 3]
```diff
@@ -0,0 +1,42 @@
+.link-container {
+    padding: 2rem;
+    width: min-content;
+}
```
Comment (greatislander) [inline src/static/css/link.css:4]: This class isn't needed.

File: src/static/css/link.css [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+a:focus {
+    color: var(--fl-linkColor, var(--link-focus-color));
+    outline: 3px solid var(--fl-linkColor, var(--link-focus-color));
+    outline-offset: 6px;
+    outline-width: 3px;
+}
+
+a:hover {
+    color: var(--fl-linkColor, var(--link-hover-color));
+}
```
Comment (greatislander) [inline src/static/css/link.css:26]: We should use the order indicated here: https://css-tricks.com/css-basics-styling-links-like-boss/#aa-style-each-link-state

**Review by greatislander — CHANGES_REQUESTED** [review]

File: src/_includes/components/link/link.njk (hunk 2 above)
Comment (greatislander) [reply src/_includes/components/link/link.njk:8]: I don't think we should have a special "link-container"— we should only wrap links which need a different background.

File: src/static/css/variables.css [hunk 5]
```diff
@@ -33,6 +33,13 @@
     --color-black-hsl: 0, 0%, 0%;
     --color-black: hsl(var(--color-black-hsl));
 
+    /* Link */
+    --link-color: var(--color-blue-600);
+    --link-active-color: var(--color-blue-600);
+    --link-active-color-bg: var(--color-blue-100);
```
Comment (greatislander) [inline src/static/css/variables.css:39]: Let's make this variable `--link-active-bg` to make it more concise.

File: src/_includes/components/link/link.njk (hunk 2 above)
Comment (chosww) [reply src/_includes/components/link/link.njk]: Thanks Ned for the suggestion, I created an applied the new utility class. But it doesn't scale too well on the examples, please let me know what you think once you see them.

Comment (chosww) [reply src/_includes/components/link/link.njk]: Oh, I forgot to add padding, please disregard this comment.

File: src/static/css/link.css [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    text-decoration: underline;
+}
+
+a:hover {
+    color: var(--fl-linkColor, var(--link-hover-color));
+}
+
+a:active {
+    background: var(--fl-bgColor, var(--link-active-bg));
+    box-shadow: 0 0 0 0.3em var(--fl-bgColor, var(--link-active-bg));
```
Comment (greatislander) [inline src/static/css/link.css:12]: I think this should probably be expressed in a unit other than `em` — probably `rem`. Unless you want a much bigger padding around an h1 link than a paragraph link which I don't think you do.

File: src/static/css/link.css [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    color: var(--fl-linkColor, var(--link-active-color));
+}
+
+a:focus {
+    color: var(--fl-linkColor, var(--link-focus-color));
+    outline: 3px solid var(--fl-linkColor, var(--link-focus-color));
```
Comment (greatislander) [inline src/static/css/link.css:18]: I'd use rem units for these too. We may also want to think about adding them to our scale.

**Review by greatislander — CHANGES_REQUESTED** [review]

**Review by greatislander — APPROVED** [review]

**Pull request merged**